src/hdf5/osw_cowvr2ioda.py

```diff
@@ -0,0 +1,402 @@
+                                                    obs_data[('windSpeed', obsErrName)])
+    obs_data[('windEastward', obsValName)] = np.array(u, dtype='float32')
+    obs_data[('windNorthward', obsValName)] = np.array(v, dtype='float32')
+    obs_data[('windEastward', obsErrName)] = np.array(u_err, dtype='float32')
+    obs_data[('windNorthward', obsErrName)] = np.array(v_err, dtype='float32')
+    # Create flags for eastward and northward wind: 1 if either speed or direction flag is > 0, otherwise 0
+    windEastward_flag = np.zeros_like(obs_data[('windSpeed', qcName)], dtype='int32')
+    windNorthward_flag = np.zeros_like(obs_data[('windSpeed', qcName)], dtype='int32')
+    flag_mask = (obs_data[('windSpeed', qcName)] > 0) | (obs_data[('windDirection', qcName)] > 0)
+    windEastward_flag[flag_mask] = 1
+    windNorthward_flag[flag_mask] = 1
+    obs_data[('windEastward', qcName)] = np.array(windEastward_flag, dtype='int32')
+    obs_data[('windNorthward', qcName)] = np.array(windNorthward_flag, dtype='int32')
+
+    obs_data = apply_gross_qc(obs_data)
+    return obs_data
+
+
+def get_WMO_satellite_ID(sensor_name):
+
+    if 'COWVR' in sensor_name:
+        WMO_sat_ID = ISS_COWVR_WMO_sat_ID
+    else:
+        WMO_sat_ID = -1
+
+    return WMO_sat_ID
+
+
+def get_global_attributes(wmo_satellite_id):
+    if wmo_satellite_id[0] == ISS_COWVR_WMO_sat_ID:
+        GlobalAttrs = {
+            "platformCommonName": "OSW_COWVR",
+            "platformLongDescription": "Ocean Surface Wind from COWVR",
+        }
+    else:
+        print(f" could not determine satellite from satelliteIdentifier: {wmo_satellite_id[0]}")
+        sys.exit()
+
+    return GlobalAttrs
+
+
+def combine_fore_aft_time(f):
+    # get the tai93 times from both fore and aft and combine into single array
+    # use fore value by default and aft value if fore is missing
+    # missing value assumed minimum (e.g. -9999.0)
+    fore_data = f['GriddedGeolocationAndFlags']['grid_time_tai93_fore'][:]
+    aft_data = f['GriddedGeolocationAndFlags']['grid_time_tai93_aft'][:]
+    missing_value = np.min(fore_data)
+    combined_data = np.copy(fore_data)
+    mask_fore_missing_aft_valid = (fore_data == missing_value) & (aft_data != missing_value)
+    combined_data[mask_fore_missing_aft_valid] = aft_data[mask_fore_missing_aft_valid]
+
+    return combined_data
+
+
+def get_mean_satellite_altitude(f):
+    # compute the mean satellite altitude
+    # this array is not the same dimension as OSW observation
+    sat_alt = f['GeolocationAndFlags']['sat_alt'][:]
+    missing_value = np.min(sat_alt)
+    mean_sat_alt = np.mean(sat_alt[sat_alt != missing_value])
+
+    return mean_sat_alt
+
+
+def get_epoch_time(obs_time_tai93):
+    # use approximate offset of 725846427s between 01Jan1970 and 01Jan1993
+    # missing value assumed minimum (e.g. -9999.0)
+    time_offset = obs_time_tai93[:].flatten()
+    missing_value = np.min(time_offset)
```

Comment:
A unit test would be helpful, especially in understanding the use of `missing_value = np.min(.)`. Isn't the missing value encoded in the attributes of the file?